### PR TITLE
feat(stresstest): initial commit for locust stress testing framework

### PR DESCRIPTION
**Pull request by anjakefala**
Part of #44.

This is not production quality (hacky code and not documented). 

Features left to add:
* request-state
* bursts

We also have to decide where to store the comet sample dumps that it references. Pushing to get eyes on it, and so folks can start playing with it. The command is `locust -f tests/test_stress.py`.

### CODE REVIEW

**Comment by anjakefala** [comment]
Also, @nritsche what should we pass as `roots` for update-datasets?

File: .travis.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ before_script:
 script:
     # Run pycodestyle on all .py files in all subfolders
     - black --check .
-    - find . -type d -o ! -name versioneer.py ! -name test_*.py ! -name _version.py -name \*.py -exec pydocstyle --convention=numpy --add-ignore=D105,D202 {} +
+    - find . -type d -o ! -name versioneer.py ! -name test_*.py ! -name DummyClient.py ! -name _version.py -name \*.py -exec pydocstyle --convention=numpy --add-ignore=D105,D202 {} +
```
Comment (nritsche) [inline .travis.yml]: this excludes `DummyClient.py` from a check by pydocstyle, a wonderfully annoying tool to nag about grammar in your docstrings:
```
./tests/DummyClient.py:1 at module level:

        D100: Missing docstring in public module

./tests/DummyClient.py:26 in public function `stopwatch`:

        D200: One-line docstring should fit on one line with quotes (found 2)

./tests/DummyClient.py:60 in public class `DummyClient`:

        D400: First line should end with a period (not ',')

./tests/DummyClient.py:67 in public method `__init__`:

        D401: First line should be in imperative mood (perhaps 'Set', not 'Sets')

./tests/DummyClient.py:85 in public method `register_start`:

        D208: Docstring is over-indented

./tests/DummyClient.py:85 in public method `register_start`:

        D214: Section is over-indented ('Parameters')

./tests/DummyClient.py:85 in public method `register_start`:

        D215: Section underline is over-indented (in section 'Parameters')

./tests/DummyClient.py:85 in public method `register_start`:

        D214: Section is over-indented ('Raises')

./tests/DummyClient.py:85 in public method `register_start`:

        D215: Section underline is over-indented (in section 'Raises')

./tests/DummyClient.py:85 in public method `register_start`:

        D214: Section is over-indented ('Parameters')

./tests/DummyClient.py:85 in public method `register_start`:

        D215: Section underline is over-indented (in section 'Parameters')

./tests/DummyClient.py:85 in public method `register_start`:

        D214: Section is over-indented ('Raises')

./tests/DummyClient.py:85 in public method `register_start`:

        D215: Section underline is over-indented (in section 'Raises')

./tests/DummyClient.py:158 in public method `register_config`:

        D401: First line should be in imperative mood (perhaps 'Register', not 'Registers')

./tests/DummyClient.py:285 in public method `request_state`:

        D400: First line should end with a period (not 'D')

./tests/DummyClient.py:355 in public method `update_datasets`:

        D401: First line should be in imperative mood (perhaps 'Query', not 'Queries')

./tests/DummyClient.py:427 in public class `DummyClientLocust`:

        D101: Missing docstring in public class
```

File: tests/DummyClient.py [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+
+        request = {"ds_id": ds_id, "ts": timestamp, "roots": roots}
+
+        response = self._send(UPDATE_DATASETS, request)
+
+        return response
+
+    @staticmethod
+    def _make_hash(data):
+        return hash(json.dumps(data, sort_keys=True))
```
Comment (nritsche) [inline tests/DummyClient.py]: The hash just changed to 128bit MurmurHash, e.g.:
```
    >>> import mmh3
    >>> h = mmh3.hash128(b"This is a long and random string.", seed=1420)
```

**Comment by anjakefala** [comment]
We should definitely not merge this branch with the redislocks stuff. My bad.

File: tests/DummyClient.py (hunk 2 above)
Comment (anjakefala) [reply tests/DummyClient.py]: So far, the ints it makes are too big and json.loads rejects them! Trying to figure it out...

Comment (anjakefala) [reply tests/DummyClient.py]: Ref: https://github.com/pandas-dev/pandas/issues/14530

Comment (nritsche) [reply tests/DummyClient.py]: I think we have to serialize a single hash into: `"{:016x}{:016x}"`

**Pull request closed without merging**